Move `solution_range` related functions to `core-primitives`

abhi3700 opened the pull request:
This PR adds a small change for use in `space-acres`:

- Move `solution_range_to_sectors` and `sectors_to_solution_range` functions to move out from `subspace-runtime` to `subspace-core-primitives`.

abhi3700 committed:
Moved functions related to solution range out of runtime to core-primitives

nazar-pc commented on crates/subspace-core-primitives/src/lib.rs:
```diff
@@ -83,6 +83,48 @@ pub const BLAKE3_HASH_SIZE: usize = 32;
+const SLOT_PROBABILITY: (u64, u64) = (1, 6);
+
+/// How many pieces one sector is supposed to contain (max)
+const MAX_PIECES_IN_SECTOR: u16 = 1000;
```
These constants should be removed and made explicit arguments in below functions

nazar-pc commented on crates/subspace-core-primitives/src/lib.rs:
```diff
@@ -83,6 +83,48 @@ pub const BLAKE3_HASH_SIZE: usize = 32;
+// Quick test to ensure functions above are the inverse of each other
+const_assert!(solution_range_to_sectors(sectors_to_solution_range(1)) == 1);
+const_assert!(solution_range_to_sectors(sectors_to_solution_range(3)) == 3);
+const_assert!(solution_range_to_sectors(sectors_to_solution_range(5)) == 5);
```
Please move this right next to `SolutionRange` type alias, it'll be easier to review in the future when we make them methods on new type rather than standalone functions and it is generally better to keep related things close to each other

abhi3700 committed:
Add 2 params to functions

Remove the constants from core-primitives & instead added as functions' arguments

nazar-pc commented on crates/subspace-core-primitives/src/lib.rs:
```diff
@@ -135,9 +135,59 @@ pub type BlockHash = [u8; 32];
+);
+const_assert!(
+    solution_range_to_sectors(sectors_to_solution_range(5, (1, 1), 1000), (1, 1), 1000) == 5
+);
```
The values for slot probability were `1, 6` and I'd prefer to leave them as such. Having `1, 1` can hide various edge-cases in math, so better to have realistic numbers in these assertions.

abhi3700 replied:
my bad 🤦‍♂️

abhi3700 committed:
Fix slot_probability values for testing

nazar-pc reviewed (approved):
Will squash and merge once CI passes
